### PR TITLE
Emit Password Change event upon user changing password

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/teleport/issues/2989

#### Description
- added UserPasswordChange related event and code
- emit UserPasswordChange event upon successful upsert
- clean up comment and code
- tested correct events were emitted

#### Screenshot
Screenshot from 2020-05-13 18-33-57

### CODE REVIEW

**Review by alex-kovoy — APPROVED** [review]

File: lib/auth/password_test.go [hunk 1]
```diff
@@ -126,13 +129,22 @@ func (s *PasswordSuite) TestTiming(c *C) {
 }
 
 func (s *PasswordSuite) TestChangePassword(c *C) {
+	s.a.IAuditLog = s.mockedAuditLog
```
Comment (awly) [inline lib/auth/password_test.go:132]: Should this live in `SetUpTest`?

File: lib/auth/password_test.go [hunk 2]
```diff
@@ -126,13 +129,22 @@ func (s *PasswordSuite) TestTiming(c *C) {
...
 	fakeClock := clockwork.NewFakeClock()
 	s.a.SetClock(fakeClock)
 	req.NewPassword = []byte("abce456")
 
+	// test change password event
+	s.mockedAuditLog.MockEmitAuditEvent = func(event events.Event, fields events.EventFields) error {
+		c.Assert(event, DeepEquals, events.UserPasswordChange)
+		c.Assert(fields[events.EventUser], Equals, "user1")
+		return nil
+	}
```
Comment (awly) [inline lib/auth/password_test.go:146]: also check that `EmitAuditEvent` was actually called

File: lib/auth/password_test.go (hunk 1 above)
Comment (kimlisa) [reply lib/auth/password_test.go]: revised

File: lib/auth/password_test.go (hunk 2 above)
Comment (kimlisa) [reply lib/auth/password_test.go]: revised

**Review by awly — APPROVED** [review]

**Pull request merged**